sshutil/templates.go

```diff
@@ -0,0 +1,195 @@
+package sshutil
+
+// Variables used to hold template data.
+const (
+	TypeKey               = "Type"
+	KeyIDKey              = "KeyID"
+	PrincipalsKey         = "Principals"
+	ExtensionsKey         = "Extensions"
+	CriticalOptionsKey    = "CriticalOptions"
+	TokenKey              = "Token"
+	InsecureKey           = "Insecure"
+	UserKey               = "User"
+	CertificateRequestKey = "CR"
+)
+
+// TemplateError represents an error in a template produced by the fail
+// function.
+type TemplateError struct {
+	Message string
+}
+
+// Error implements the error interface and returns the error string when a
+// template executes the `fail "message"` function.
+func (e *TemplateError) Error() string {
+	return e.Message
+}
+
+// TemplateData is an alias for map[string]interface{}. It represents the data
+// passed to the templates.
+type TemplateData map[string]interface{}
+
+// CreateTemplateData returns a TemplateData with the given certificate type,
+// key id, principals, and the default extensions.
+func CreateTemplateData(ct CertType, keyID string, principals []string) TemplateData {
+	return TemplateData{
+		TypeKey:       ct.String(),
```

Comment:
I didn't know that you could use variable names as keys when defining a map this way. cool.